source/reactor/inc/reactor/EventHandlers.h

```diff
@@ -0,0 +1,68 @@
+#ifndef Reactor_EventHandlers_H
+#define Reactor_EventHandlers_H
+
+#include <functional>
+
+namespace u {
+namespace reactor {
+class EventLoop;
+
+/**
+ * @brief Event handler function definition.
+ *
+ * Parameters to the event handler function are:
+ * - The file descriptor of the OS resource where the event was originated at. Event handler can
+ * perform, for example, accept, read, write and close operations on the file descriptor.
+ * - The event loop where it was caught. Event handler can install new handlers, remove existing
+ * handlers and even ask for loop termination.
+ */
+using EventHandler = std::function<void(int, EventLoop *)>;
+
+/**
+ * Set of event handler functions, one function to handle each possible event happened.
+ * Handlers are grouped into a struct to facilitate passing them as a single parameter to the install and
+ * remove handlers functions of the event loop.
+ * Event handlers are optional and if not set for a given event, that event will be silently
+ * ignored.
+ */
+struct EventHandlers {
```

Comment:
For me Event-Handlers sounds like _many_ handlers for the _same_ event.
For me it is natural to call this: Event**s**-Handler as it is a single handler for few events.

Reply:
Actually they are a set of handlers, one for each different event that might get fired (read, write, error and timeout).